### PR TITLE
#759 mailto refactoring

### PR DESCRIPTION
**Pull request by kojoty**
#759 + full mailto refactoring
Now user2user mails are send in HTML and directly on user request - without queued through the DB.

### CODE REVIEW

**Comment by kojoty** [comment]
@harrieklomp, @andrixnet there are small changes in translation files

**Comment by harrieklomp** [comment]
No translations needed for Dutch translations.

**Comment by kojoty** [comment]
@harrieklomp, I'm sure that at least 'mail_replyAllowed' is a new key in translations.

**Comment by harrieklomp** [comment]
You are correct. I did not see that. This line was not added to Dutch translation file.
I will add this line and translate them after commit :-)

File: mailto.php [hunk 1]
```diff
@@ -1,123 +1,145 @@
...
+
+
+            $text = mb_ereg_replace('{{text}}', $text, $send_emailaddress == 1 ? $mailtext_email : $mailtext_anonymous);
+            $text = mb_ereg_replace('{from_userid}', $usr["userid"], $text);
+            $text = mb_ereg_replace('{from_email}', $from_email, $text);
+            $text = mb_ereg_replace('{from_username}', $from_username, $text);
+            $text = mb_ereg_replace('{to_email}', $to_email, $text);
+            $text = mb_ereg_replace('{to_username}', $to_username, $text);
+
+            XDb::xSql("INSERT INTO `email_user`
```
Comment (mzylowski) [inline mailto.php]: So we decided to still keep mails in database?

File: Utils/Email/EmailFormatter.php [hunk 2]
```diff
@@ -50,4 +57,20 @@ public function addFooterAndHeader($username, $automatically=true) {
 
         $this->emailContent = $header->getEmailContent() . $this->emailContent . $footer->getEmailContent();
     }
+
+    /**
+     * this method provide translation in email templates
+     * It looks for {{transl_key}} phases and replace it to proper translation string
+     */
+    private function loadTranslations()
```
Comment (mzylowski) [inline Utils/Email/EmailFormatter.php]: Very good idea. Definitely better than https://github.com/opencaching/opencaching-pl/pull/915/commits/f3b664c0de75cd584d6209025a64164c6bc39b4c 
I will use this in all emails in #915

File: mailto.php (hunk 1 above)
Comment (kojoty) [inline mailto.php]: yes and no - this line will be removed (there is an exit above), but I decided to store a trace of every message in db - we can discuss it separately in future...

**Comment by andrixnet** [comment]
Feel free to merge. I have more RO strings pending and will integrate these as well. (assuming EN is properly updated)

**Comment by mzylowski** [comment]
@kojoty after #915 you have some trivial conflicts to fix.

**Comment by mzylowski** [comment]
Why we still waiting for merge?

**Comment by mzylowski** [comment]
@kojoty you will merge this soon? I'm waiting for your changes from this PR in Email Class ;] 

Or I just should cherry pick some your commits and continue my work.

**Comment by kojoty** [comment]
Czarodziej, 

postaram się zrobić to dziś - tam jest jeszcze taki problem (zdałem sobie z tego sprawę), że ja usuwam skrypt wołany z crona do wysyłki poczty przez bazę - tymczasem nie wiem jaka jest konfiguracja crona na innych serwerach i nie wiem czy przypadkiem nie spowoduje to, że jakaś wiadomość "utknie" w bazie

myslalem ze zrobię jeszcze poprawkę w kodzie, ale to chyba tak marginalne ze tylko napiszę kolegom, zeby sprawdzili czy nie ma tego problemu w ich bazie

juz dawno bym to wszystko zrobił ale jakoś ostatnio nie miałem czasu (problemy z serwerem + mam w domu remont i inne takie)

podsumowując: dziś postaram się to mergować!

**Comment by kojoty** [comment]
@harrieklomp, @andrixnet, @deg-pl 
please note that now user emails are sent without queued it in DB so cron line for    
**util.sec/email_sender/email_sender.php email_sender.html**
is not needed anymore.

**Comment by harrieklomp** [comment]
Thanks. 
Switched it off with a "#" on oc.nl

**Comment by deg-pl** [comment]
Done @ OC UK

**Comment by andrixnet** [comment]
Done @ OCRO.

P.S. 
Can this change affect user interraction? Or mail server overload?

**Comment by kojoty** [comment]
Generally speaking, I don't think so. If you use linux and send emails by local mta (postfix of different) php adds the new email to the local queue in the same way as when it is taken from DB. If we would like to use other solution (for example sends emails by remote server) then probably queued in db is reasonable.

Additionally please note that all other emails are sent directly (like user2user messages now) and we don't have any problems with it.

**Pull request merged**